### PR TITLE
Removes Why Did We Create Odo and cleans up README

### PR DESCRIPTION
**Pull request by cdrage**
Removes the "Why Did we Create Odo" section and cleans up the README so
it's more concise / smaller.

I believe that there's *too much* text and multiple paragraphs that iterate
over the same concept (Odo is fast, it reduces development time, etc.).

Having a few sentences / list of features is more than enough.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,38 +4,36 @@
...
 - [Glossary](#glossary)
 - [CLI Structure](#cli-structure)
 
 
-## What is Odo?
+## Odo
 
 OpenShift Do (Odo) is a CLI tool for developers who are writing, building, and deploying applications on OpenShift. With Odo, developers get an opinionated CLI tool that supports fast, iterative development which abstracts away Kubernetes and OpenShift concepts, thus allowing them to focus on what's most important to them: code.
 
+Odo was created to improve the developer experience with OpenShift. Existing tools such as `oc` are more operations-focused and requires a depp-understanding of Kubernetes and OpenShift concepts. Odo is designed to be simple and concise so you may focus on coding rather than how to deploy your application. Since Odo can build and deploy your code to your cluster immediately after you save you changes, you benefit from instant feedback and can thus validate your changes in real-time.
```
Comment (metacosm) [inline README.md]: Typo: depp->deep. Also I don't think that `deep understanding` is hyphenated.

File: README.md [hunk 2]
```diff
@@ -4,38 +4,36 @@
...
-
-## Who is Odo For?
-
-Odo is for developers! If you are a developer who is creating, building, or maintaining applications on OpenShift, Odo was designed for you. Odo supports multiple languages and frameworks, so whether you're using Node.js, Java, Ruby, Perl, PHP, or Python, the goal of Odo is to improve your development experience and to support fast, iterative development of applications for OpenShift. 
+## Features
 
-Odo is *not* a replacement for the standard `oc` CLI tool, and it is not intended to be used by operations, security, or release management teams.
+- Designed for fast, iterative developmeny cycles
```
Comment (metacosm) [inline README.md]: developmeny -> development

File: README.md (hunk 1 above)
Comment (metacosm) [reply README.md]: I would still mention `odo` core concepts here (project, application, components).

File: README.md [hunk 3]
```diff
@@ -4,38 +4,36 @@
...
+- No server needed. 100% OpenShift API communication
```
Comment (metacosm) [inline README.md]: Not clear.

File: README.md [hunk 4]
```diff
@@ -4,38 +4,36 @@
...
+- Supports multiple languages and frameworks such as Node.js, Java, Ruby, Perl, PHP and Python
+- Detect changes to your local code and deploy automatically with `odo watch`
+- List all available components and services from your OpenShift cluster
+- Support for linking and multiple deployments
```
Comment (metacosm) [inline README.md]: Not clear. Also not a fan of feature list because they tend to get outdated.

**Comment by cdrage** [comment]
@metacosm Updated the README to address your comments :+1:

**Comment by metacosm** [comment]
@cdrage did you force push? The UI fails to show updates. Looking at the files directly, LGTM, though.

**Pull request merged**